src/__tests__/parser.test.ts

```diff
@@ -29,6 +34,15 @@ describe("Parsing CSVs ", () => {
     listedTokens = Array.from(tokenList.keys());
     listedToken = tokenList.get(listedTokens[0]);
   });
+
+  it("should throw errors for invalid CSVs", async () => {
+    // thins csv contains more values than headers in row1
+    const invalidCSV = "head1,header2\nvalue1,value2,value3";
+    expect(parseCSV(invalidCSV, tokenList)).to.be.rejectedWith(
+      "column header mismatch expected: 2 columns got: 3"
+    );
+  });
```

Comment:
chaiAsPromised gives the rejectedWith method. Otherwise rejection cannot really be checked with chai as far as I understood the docs. ChaiAsPromised is a official plugin though. Also I added the typings for both chai and chaiAsPromised such that the typescript compiler now knows what is possible to call.